src/RequestClientWrapper.ts

```diff
@@ -97,21 +100,26 @@ export default class RequestClientWrapper {
     );
   }
 
-  private makeRequest(
+  private async makeRequest(
     method: HttpRequestMethod,
     callStart: Date,
     numberOfCalls: number,
     url: string,
     options: AxiosRequestConfig,
     callback: RequestCallback
-  ): void {
+  ): Promise<void> {
     const callDetails: CallDetails = { options, callStart, callCountForThisRequest: numberOfCalls };
     options.baseURL = this.requestClientWrapperConfiguration.endpoint;
     if (options.headers === undefined) {
       options.headers = {};
     }
-    options.headers['X-Channel-Ape-Authorization-Token'] = this.requestClientWrapperConfiguration.session;
-    options.headers['Content-Type'] = 'application/json';
+    if (this.isItJwtToken(this.requestClientWrapperConfiguration.session)) {
```

Comment:
Hmmm, regex can be expensive, especially if we are trying to do the match on every request. What if we run this on instantiation of the sdk with the session and store the boolean isJwt as a private field and we just check that. This i think would be a huge performance gain on these checks.

Reply:
@ryankazokas Not sure what you are referring to here. The function isItJwtToken uses a simple JSON.parse.

Reply:
ah right. But still i imagine that has some overhead over just checking a boolean if there isn't regex within json.parse. I'd still update this to only do that try/catch once.

Reply:
@ryankazokas I changed this to be determined when constructed and added a class property. It will only get used once now.